kratos/expression/entity_domain_size_expression_io.cpp

```diff
@@ -0,0 +1,84 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Suneth Warnakulasuriya
+//
+
+// System incldues
+
+// Project includes
+#include "expression/literal_flat_expression.h"
+#include "utilities/parallel_utilities.h"
+
+// Include base h
+#include "entity_domain_size_expression_io.h"
+
+namespace Kratos {
+
+EntityDomainSizeExpressionIO::EntityDomainSizeExpressionInput::EntityDomainSizeExpressionInput(
+    const ModelPart& rModelPart,
+    const ContainerType& rContainerType)
+    : mrModelPart(rModelPart),
+      mContainerType(rContainerType)
+{
+}
+
+Expression::Pointer EntityDomainSizeExpressionIO::EntityDomainSizeExpressionInput::Execute() const
+{
+    const auto& r_mesh = mrModelPart.GetCommunicator().LocalMesh();
+
+    LiteralFlatExpression<double>::Pointer expression;
+
+    switch (mContainerType) {
+        case ContainerType::ConditionNonHistorical:
+            expression = LiteralFlatExpression<double>::Create(r_mesh.NumberOfConditions(), {});
+            IndexPartition<IndexType>(r_mesh.NumberOfConditions()).for_each([&r_mesh, &expression](const IndexType Index) {
+                const auto& r_condition = *(r_mesh.ConditionsBegin() + Index);
+                *(expression->begin() + Index) = r_condition.GetGeometry().DomainSize();
+            });
+            break;
+        case ContainerType::ElementNonHistorical:
+            expression = LiteralFlatExpression<double>::Create(r_mesh.NumberOfElements(), {});
+            IndexPartition<IndexType>(r_mesh.NumberOfElements()).for_each([&r_mesh, &expression](const IndexType Index) {
+                const auto& r_element = *(r_mesh.ElementsBegin() + Index);
+                *(expression->begin() + Index) = r_element.GetGeometry().DomainSize();
+            });
+            break;
+        default:
+            KRATOS_ERROR << "EntityDomainSizeExpressionInput only supports ConditionNonHistorical and ElementNonHistorical container types.";
+            break;
+    }
```

Comment:
seems like a job for [proxies](https://github.com/KratosMultiphysics/Kratos/blob/fbc41845e9059b8457c9fa44dd0d2457a1c48931/kratos/utilities/proxies.h)

```cpp
template <Globals::DataLocation TContainerType>
Expression::Pointer ReadDomainSizes(const ModelPart& rModelPart)
{
    LiteralFlatExpression<double>::Pointer p_expression;

    auto container = MakeProxy<TContainerType>(rModelPart);
    IndexPartition<IndexType>(container.size()).for_each([&p_expression, &container] (IndexType iEntity) {
        const auto& r_entity = container[i_entity].GetEntity();
        *(p_expression->begin() + iEntity) = r_entity.DomainSize();
    });

    return p_expression;
}

auto p_expression = ReadDomainSizes<Globals::DataLocation::Element>(rModelPart); // <== gets the domain sizes of elements
auto p_expression = ReadDomainSizes<Globals::DataLocation::Condition>(rModelPart); // <== gets the domain sizes of conditions
```

Reply:
This won't work since, I need to iterate over only the local entities. This  passes through all the local and ghost entities.